include/board.h

```diff
@@ -2,9 +2,9 @@
 #define BOARD_H_INCLUDED
 
 typedef struct {
-  int rows;
-  int cols;
-  int *cells;
+  unsigned short rows;
```

Comment:
Might be left unsigned int.
Why did you switch to short?

Reply:
That was just to save a few Bytes of memory.
I get that it doesn't make much of a difference either way.

Reply:
If we switch them to `unsigned`, we would have to switch the loop iterators to `unsigned`, wouldn't we?
Otherwise we could have the case were the iterator cannot grow as big as the parameter it follows.

Reply:
I even considered switching `rows` and `cols` to `unsigned char` and check the user input to be smaller than 255.
Again, minor space gains.